python/instrumentation/openinference-instrumentation-haystack/src/openinference/instrumentation/haystack/_wrappers.py

```diff
@@ -0,0 +1,207 @@
+                            }
+                        )
+                    )
+
+                    i = 0
+                    for document in response['documents']:
+                        attributes[f"{SpanAttributes.RETRIEVAL_DOCUMENTS}.{i}.{DocumentAttributes.DOCUMENT_CONTENT}"] = document.content
+                        attributes[f"{SpanAttributes.RETRIEVAL_DOCUMENTS}.{i}.{DocumentAttributes.DOCUMENT_ID}"] = document.id
+                        attributes[f"{SpanAttributes.RETRIEVAL_DOCUMENTS}.{i}.{DocumentAttributes.DOCUMENT_SCORE}"] = document.score
+                        attributes[f"{SpanAttributes.RETRIEVAL_DOCUMENTS}.{i}.{DocumentAttributes.DOCUMENT_METADATA}"] = safe_json_dumps(document.meta)
+                        i += 1
+
+                case 'prompt_builder':
+                    attributes = dict(
+                        _flatten(
+                            {
+                                SpanAttributes.OPENINFERENCE_SPAN_KIND : OpenInferenceSpanKindValues.CHAIN,
+                                SpanAttributes.LLM_PROMPT_TEMPLATE: instance.graph.nodes._nodes["prompt_builder"]["instance"]._template_string,
+                                SpanAttributes.INPUT_VALUE: attributes['parameters'],
+                                SpanAttributes.INPUT_MIME_TYPE: OpenInferenceMimeTypeValues.JSON,
+                                SpanAttributes.RETRIEVAL_DOCUMENTS: safe_json_dumps(invocation_parameters['documents']),
+                            }
+                        )
+                    )
+
+            span.set_attributes(attributes)
+
+        return response
+
+class _PipelineWrapper(_WithTracer):
+    """
+    Wrapper for the pipeline processing
+    Captures all calls to the pipeline
+    """
+    def __call__(
+            self,
+            wrapped: Callable[..., Any],
+            instance: Any,
+            args: Tuple[Any, ...],
+            kwargs: Mapping[str, Any],
+    ) -> Any:
+        if context_api.get_value(context_api._SUPPRESS_INSTRUMENTATION_KEY):
+            return wrapped(*args, **kwargs)
+
+        # Prepare invocation parameters by merging args and kwargs
+        invocation_parameters = {}
+        for arg in args:
+            if arg and isinstance(arg, dict):
+                invocation_parameters.update(arg)
+        invocation_parameters.update(kwargs)
+
+        span_name = "Pipeline"
+        with self._tracer.start_as_current_span(
+                span_name,
+                attributes={}
+        ) as span:
+            span.set_attributes(dict(get_attributes_from_context()))
+            try:
+                response = wrapped(*args, **kwargs)
+            except Exception as exception:
+                span.set_status(trace_api.Status(trace_api.StatusCode.ERROR, str(exception)))
+                span.record_exception(exception)
+                raise
+            span.set_status(trace_api.StatusCode.OK)
+
+            attributes = dict(
```

Comment:
same comment re setting the invocation parameters before the wrapped function is executed